lib/logstash/gemfile.rb

```diff
@@ -0,0 +1,173 @@
+module LogStash
+
+  class GemfileError < StandardError; end
+
+  class Gemfile
+    attr_accessor :gemset
+
+    HEADER = \
+      "# This is a Logstash generated Gemfile.\n" + \
+      "# If you modify this file manually all comments and formatting will be lost.\n\n"
+
+    # @params io [IO] any IO object that supports read, write, truncate, rewind
+    def initialize(io)
+      @io = io
+      @gemset = nil
+    end
+
+    def load
+      @gemset ||= Dsl.parse(@io.read)
+      self
+    end
+
+    def save
+      raise(GemfileError, "a Gemfile must first be loaded") unless @gemset
+      @io.truncate(0)
```

Comment:
we should consider using file locks (flock(2)) for saving/loading to prevent concurrent plugin installers from doing battle.